synapse/module_api/__init__.py

```diff
@@ -1547,6 +1547,37 @@ async def create_room(
 
         return room_id_and_alias["room_id"], room_id_and_alias.get("room_alias", None)
 
+    async def set_displayname(
+        self,
+        target_user: UserID,
+        requester_user_id: UserID,
+        new_displayname: str,
+        deactivation: bool = False,
+    ) -> None:
+        """Sets a user's display name.
+
+        Added in Synapse v1.74.0.
+
+        Args:
+            target_user:
+                The user whose display name is to be changed.
+            requester:
+                The user attempting to make this change.
+            new_displayname:
+                The new display name to give the user.
+            deactivation:
+                Whether this change was made while deactivating the user.
+        """
+        requester = create_requester(requester_user_id)
+        by_admin = await self.is_user_admin(requester_user_id.to_string())
+        await self._hs.get_profile_handler().set_displayname(
+            target_user=target_user,
+            requester=requester,
+            new_displayname=new_displayname,
+            by_admin=by_admin,
```

Comment:
We usually consider that a module always acts as a server admin, because setting it up on the server requires admin privileges, has privileged access to the homeserver's features. So in this case I think it would be better to just set `by_admin` to `True`, build the requester from `target_user` (maybe even rename this argument `user_id`?) and drop the `requester_user_id` argument.